src/canopy/llm/openai.py

```diff
@@ -94,9 +94,11 @@ def chat_completion(self,
         model_params_dict.update(
             model_params or {}
         )
+        model = model_params.get("model", self.model_name) if model_params else self.model_name
+        model_params_dict = {k: v for k, v in model_params_dict.items() if k != 'model' and v is not None} if model_params else None
```

Comment:
I think I more complete solution here will probably need to make sure we're only passing params to OpenAI that their API supports